src/analyses/wrapperFunctionAnalysis.ml

```diff
@@ -131,8 +131,19 @@ module MallocWrapper : MCPSpec = struct
       CilType.Location.show loc
 
     let name_varinfo (t, node, c) =
-      Format.asprintf "(alloc@sid:%s@tid:%s(#%s))" (Node.show_id node) (ThreadLifted.show t) (UniqueCount.show c)
-
+        let uniq_count =
+          if UniqueCount.is_top c then
+            Format.dprintf ""
```

Comment:
If the uniqueness counter is activated and some things are kept apart, would it not be useful to know that this is

`alloc@...#T` (which means it's not `alloc@...#0`). If nothing is kept apart, I agree that this is indeed superfluous.

Reply:
Useful to know what is?

This is completely consistent whether or not the counter limit is greater than zero: the variable without `#i` is the non-unique one, those with `#i` are the unique ones. We don't need two different representations for the same thing.

We don't need Goblint to tell the user in every warning what analysis the user activated. They activated those things themselves in the configuration they wrote.

Reply:
I would want to follow the principle of least surprise. If I enable an option to keep `n` allocs apart, I want the system to tell me whether it was able to assign a unique counter to a given heap variable. Otherwise, I have to do some sort of inferring what it means for the system to leave out the `#i`.

Imagine for example I have a huge program and I want to inspect the warning at one program point. I turn on the uniqueness counter, and nothing changes about the warning. Was it that I specified the option in the wrong file? Did I load an old result? If I now see `#top` I know that my option had some effect.

Reply:
> Was it that I specified the option in the wrong file? Did I load an old result? If I now see `#top` I know that my option had some effect.

That logic doesn't apply to most of our options though. Making analyses path-sensitive, enabling int domain refinement, enabling widening thresholds, adding malloc or thread wrappers, etc.
The goal of this PR is not to fix the new issue you bring up. Nor does keeping the `#top` fix users accidentally looking at some wrong path for configuration or results.

Reply:
Maybe we should discuss this at the GobCon with everyone. I feel the proposed change here makes it harder to understand and interpret the result rather than easier.